provider/docker.go

```diff
@@ -82,56 +124,96 @@ func (provider *Docker) Provide(configurationChan chan<- types.ConfigMessage, po
 			ctx := context.Background()
 			version, err := dockerClient.ServerVersion(ctx)
 			log.Debugf("Docker connection established with docker %s (API %s)", version.Version, version.APIVersion)
-			containers, err := listContainers(ctx, dockerClient)
-			if err != nil {
-				log.Errorf("Failed to list containers for docker, error %s", err)
-				return err
+			var dockerDataList []DockerData
+			if provider.SwarmMode {
```

Comment:
Nit but we could extract this in a method (`listDockerData`) :stuck_out_tongue_closed_eyes: